From c5672a992ad4915b1d84ad8aa1c04a100809b20c Mon Sep 17 00:00:00 2001
From: akallabeth <akallabeth@posteo.net>
Date: Tue, 3 May 2022 20:46:15 +0200
Subject: [PATCH] Unified neon source options

(cherry picked from commit bfdbe180297f475fe585a1ec99e5f7ca22ab9bc2)
---
 libfreerdp/CMakeLists.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/libfreerdp/CMakeLists.txt b/libfreerdp/CMakeLists.txt
index 0b55388168e..a259b68cc30 100644
--- a/libfreerdp/CMakeLists.txt
+++ b/libfreerdp/CMakeLists.txt
@@ -313,7 +313,7 @@ freerdp_definition_add(-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
 
 ### IPP Variable debugging
 if(WITH_IPP)
-    if(CMAKE_COMPILER_IS_GNUCC)
+    if(CMAKE_COMPILER_IS_GNUCC OR ${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
         foreach(INCLDIR ${IPP_INCLUDE_DIRS})
             set(OPTIMIZATION "${OPTIMIZATION} -I${INCLDIR}")
         endforeach(INCLDIR)
@@ -335,9 +335,11 @@ if(WITH_SSE2)
             PROPERTIES COMPILE_FLAGS "${OPTIMIZATION} /arch:SSE2")
     endif()
 elseif(WITH_NEON)
-    if(CMAKE_COMPILER_IS_GNUCC)
-        set_source_files_properties(${PRIMITIVES_OPT_SRCS}
-            PROPERTIES COMPILE_FLAGS "${OPTIMIZATION} -mfpu=neon")
+    if(CMAKE_COMPILER_IS_GNUCC OR ${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
+        if (NOT MSVC_ARM64 AND NOT ARCH_ARM64)
+            set_source_files_properties(${PRIMITIVES_OPT_SRCS}
+                PROPERTIES COMPILE_FLAGS "${OPTIMIZATION} -mfpu=neon")
+        endif()
     endif()
     # TODO: Add MSVC equivalent
 endif()
